usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -370,8 +368,68 @@ def fetch_business_categories_by_legal_entity_id(legal_entity_id):
     return []
 
 
+def fetch_all_cfda_details(award):
+    queryset = TransactionFABS.objects.filter(transaction__award_id=award["id"],).values(
+        "cfda_number", "federal_action_obligation", "non_federal_funding_amount", "total_funding_amount")
+    cfdas = {}
+    for item in queryset:
+        # sometimes the transactions data has the trailing 0 in the CFDA number truncated, this adds it back
+        num = re.sub(r"^\d*\.\d\d$", "\\g<0>0", item["cfda_number"])
```

Comment:
```suggestion
        num = re.sub(r"^\d{3}\.\d\d$", "\\g<0>0", item["cfda_number"])
```
I vaguely recall you telling me this was always three...

Reply:
it's always three after the decimal point, hence the `\d\d$`

Reply:
ok. Is there any concern of two (or all) trailing zeros being truncated, or is the second decimal place always >0 in a CFDA?

Reply:
yup, that is a risk, which i will now go fix

Reply:
Ohh, and what about leading zeros?

Reply:
smallest CFDA number is 10.001, largest is 98.012, no leading zeroes to be found